src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -675,6 +675,14 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (isAllIndices(aliasesOrIndices)) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new IndexMissingException(new Index("_all"));
+            }
+            else return Strings.EMPTY_ARRAY;
```

Comment:
regardless of the below comment, use brackets here please and move the `else` the to the previous line.